subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -367,18 +415,62 @@ private void printSection(StyledTextOutput out, Section extraDetail, int depth)
         }
 
         private void printVariantDetails(StyledTextOutput out, RenderableDependency dependency) {
```

Comment:
When I look at this with a multi-project build, I don't see a selected variant for project dependencies.  I used `gradle init` and picked the multi-project application project.